Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1269,6 +1269,53 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return handleEffectSpecifiers(node)
   }
 
+  public override func visit(_ node: TypeInheritanceClauseSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let unexpected = node.unexpectedBeforeColon,
+      let leftParen = unexpected.tokens(satisfying: { $0.tokenKind == .leftParen }).first
+      {
+        
+        var handledNodes: [SyntaxIdentifier] = [
+            leftParen.id, node.colon.id,
+        ]
+        
+        var changes: [FixIt.MultiNodeChange] = [
+            .makePresent(node.colon, trailingTrivia: .space),
+            .makeMissing(unexpected),
+        ]
+        
+        var replaceTokens = [leftParen]
+        
+        if let rightParen = node.unexpectedAfterInheritedTypeCollection?.tokens(satisfying: { $0.tokenKind == .rightParen }).first,
+           let lastToken = node.inheritedTypeCollection.lastToken(viewMode: .sourceAccurate)
+        {
+            handledNodes += [rightParen.id]
+            changes += [
+                .makeMissing(rightParen),
+                FixIt.MultiNodeChange(.replaceTrailingTrivia(token: lastToken, newTrivia: .space)),
```

Comment:
I think adding the space is only necessary if the last token doesn’t have any trailing trivia. Otherwise, we might be overriding existing trailing trivia, which could be comments the user wants to keep. If you want, you can add a convenience function like `ensuringTrailingSpace` to `MultiNodeChange` for this. We might want to use that in other, existing cases as well, I haven’t looked through the code though.

Reply:
We will use that in other places. I found many examples in #1549. So I was already there thinking about adding a method somewhere we can do this easily.

Reply:
Something like this?

Reply:
I think this might be fixed with my upcoming changes to BasicFormat (there’s one final PR that’s not up yet). I would like to see if we actually need the explicit insertion of trivia once that’s merged.